drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -836,27 +834,28 @@ static void nrf5_receive_at(uint32_t start, uint32_t duration, uint8_t channel,
 
 static void nrf5_config_csl_period(uint16_t period)
 {
-	nrf_802154_receive_at_cancel(DRX_SLOT_PH);
-	nrf_802154_receive_at_cancel(DRX_SLOT_RX);
-
 	nrf_802154_csl_writer_period_set(period);
 
-	/* A placeholder reception window is scheduled so that the radio driver is able to inject
-	 * the proper CSL Phase in the transmitted CSL Information Elements.
+	/* Update the CSL anchor time to match the nearest requested CSL window, so that
+	 * the proper CSL Phase in the transmitted CSL Information Elements can be injected.
 	 */
 	if (period > 0) {
-		nrf5_receive_at(nrf5_data.csl_rx_time, PH_DURATION, nrf_802154_channel_get(),
-				DRX_SLOT_PH);
+		nrf_802154_csl_writer_anchor_time_set(nrf5_data.csl_rx_time);
 	}
 }
 
 static void nrf5_schedule_rx(uint8_t channel, uint32_t start, uint32_t duration)
 {
 	nrf5_receive_at(start, duration, channel, DRX_SLOT_RX);
 
-	/* The placeholder reception window is rescheduled for the next period */
-	nrf_802154_receive_at_cancel(DRX_SLOT_PH);
-	nrf5_receive_at(nrf5_data.csl_rx_time, PH_DURATION, channel, DRX_SLOT_PH);
+	/* Update the CSL anchor time to match the nearest requested CSL window, so that
+	 * the proper CSL Phase in the transmitted CSL Information Elements can be injected.
+     *
+	 * Note that even if this function is not called before the next CSL window starts
```

Comment:
This comment is a bit misleading, because it suggests that no matter if this function is called or not, the reception windows will be scheduled in the same way. I'd suggest rephrasing it to explain the difference between these two situations:
```
* The following call updates the CSL anchor time used to calculate CSL phase. If it is not executed before
* the nearest CSL window starts, the CSL phase is going to be calculated based on the previously set
* CSL anchor time. Although less accurate than the latest one, the previous CSL anchor time should
* still be accurate enough to allow for robust CSL reception.
```

Reply:
Ok, I wasn't precise enough, "this function" was supposed to mean `nrf5_schedule_rx`, not `nrf_802154_csl_writer_anchor_time_set` I changed a commit a little bit.

By the way note that the change won't fix the fact, that the CSL window is not being scheduled - it will only protect against garbage values in the CSL phase